dali/ft/daftformat.cpp

```diff
@@ -143,8 +143,11 @@ void CPartitioner::commonCalcPartitions()
         //Don't add an empty block on the start of the this chunk to transfer.
         if ((split != firstSplit) || (inputOffset != startInputOffset))
         {
-            results.append(*new PartitionPoint(whichInput, split-1, startInputOffset-thisOffset+thisHeaderSize, inputOffset - startInputOffset - cursor.trimLength, cursor.outputOffset-startOutputOffset));
-            JSON_DBGLOG("commonCalcPartitions: startInputOffset:%lld, inputOffset: %lld, startOutputOffset: %lld, cursor.outputOffset: %lld",startInputOffset ,inputOffset, startOutputOffset, cursor.outputOffset);
+            // To avoid the first partition's whichOutput value becomes -1 (which means invalid)
+            unsigned whichOutput = (split > 0 ? split-1 : split);
```

Comment:
when does it enter this block, such that split is 0?
I thought the if (split != firstSplit) covered that case?
Perhaps comment need extending to explain.

Reply:
After some tracing and debugging I find out when the split == 0 the partition out file index becomes “-1” what means invalid/not defined file. For usual files (with multiple records what can distribute among the target parts) the split moving from first split to the last one during the partition creation, but in a special occasion e.g. when there is only one record for a multiple target parts environment, the scan can reach the end of the record (and the input file) and it is still in the split ==0 state.